### PR TITLE
Prac 05 Code Review Request

### PR DESCRIPTION
**Pull request by Belchy06**
Please check formatting, variable names and logic @jai-wilson

### CODE REVIEW

File: prac_04/repeated_string_printer.py [hunk 1]
```diff
@@ -39,6 +41,7 @@ def print_repeated_strings(repeated_strings):
 def main():
     print_instructions()
     string_sequence = get_string_sequence()
+    #string_sequence = ["hello", "world", "hello", "world", ".", ""]
     repeated_strings = find_repeated_strings(string_sequence)
     print_repeated_strings(repeated_strings)
```
Comment (Jai-Wilson) [inline prac_04/repeated_string_printer.py:47]: Great use of functions presented in the course so far. Maybe add some comments to aid a reviewer reading the program?

File: prac_05/state_names.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+"""
+
+CODE_TO_NAME = {
+    "QLD": "Queensland",
+    "NSW": "New South Wales",
+    "NT": "Northern Territory",
+    "WA": "Western Australia",
+    "ACT": "Australian Capital Territory",
+    "VIC": "Victoria",
+    "TAS": "Tasmania"
```
Comment (Jai-Wilson) [inline prac_05/state_names.py:14]: Great spacing, allows reading to be a lot easier

File: prac_05/word_occurences.py [hunk 3]
```diff
@@ -3,14 +3,34 @@
...
-if input_string != "":
+def main():
+    user_input = get_input_string()
+    word_occurrences = find_occurrences(user_input)
+    print_occurrences(word_occurrences)
+
+
+def get_input_string():
+    input_string = input("Enter your string: ").lower().split()
+    while input_string == "":
```
Comment (Jai-Wilson) [inline prac_05/word_occurences.py:15]: Is this supposed to be an error handling function if the user enters nothing?

File: prac_05/word_occurences.py [hunk 4]
```diff
@@ -3,14 +3,34 @@
...
         except KeyError:
             word_occurrences[string] = 1
+    return word_occurrences
+
+
+def print_occurrences(word_occurrences):
+    string_spacing = len(max(word_occurrences, key=len))
+    for key, value in word_occurrences.items():
+        print("{:{}}: {}".format(key, string_spacing, value))
+
```
Comment (Jai-Wilson) [inline prac_05/word_occurences.py:35]: Great use of separate functions, easy to follow the flow of the program

File: prac_05/emails.py [hunk 5]
```diff
@@ -2,24 +2,57 @@
...
+            valid_response = True
+        elif response[0] == 'n':
+            specified_name = input("Name: ")
+            while specified_name == "":
+                specified_name = input("Name: ")
+            username = specified_name
+            valid_response = True
+        else:
+            response = input("Is your name {}? (y/n): ".format(original_name)).lower()
+    return username
```
Comment (Jai-Wilson) [inline prac_05/emails.py:49]: Great use of function, allows for many possible user entries

File: prac_04/repeated_string_printer.py (hunk 1 above)
Comment (Belchy06) [reply prac_04/repeated_string_printer.py:47]: Not a bad idea to change the variable names. Was having issues tryna sort our variable names

**Review by Belchy06 — COMMENTED** [review]
Looking good. Can you approve the PR @Jai-Wilson

**Review by Jai-Wilson — APPROVED** [review]

**Pull request merged**